UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Extensions/CallbackUriExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace UKHO.ExchangeSetService.API.Extensions
+{
+    public static class CallbackUriExtensions
+    {
+        public static bool IsValidCallbackUri(this string callbackUri)
+        {
+            try
+            {
+                Uri baseUri = new Uri(callbackUri);
+                return (baseUri.Scheme == Uri.UriSchemeHttp || baseUri.Scheme == Uri.UriSchemeHttps);
```

Comment:
This should be raised in your threat modeling.,

We should not be calling HTTP

Reply:
Changes done to avoid calling HTTP and it will only allow HTTPS.